components/nexus-httpclient/src/main/java/org/sonatype/nexus/httpclient/config/NonProxyHostsValidator.java

```diff
@@ -42,16 +45,15 @@ public boolean isValid(final String[] values, final ConstraintValidatorContext c
    */
   private boolean isValid(final String value) {
     // A value should be a non-empty string optionally prefixed or suffixed with an asterisk
-    if (Strings.isNullOrEmpty(value)) {
-      // must be non-empty
+    if (Strings2.isBlank(value)) {
+      // must be non-empty, non-blank
       return false;
     }
     if (value.contains("|")) {
       // must not contain | separator (used to separate multiple values in system properties)
       return false;
     }
-    if (value.contains("*") && !(value.startsWith("*") || value.endsWith("*"))) {
-      // if contains asterisk, it must be at beginning or end only
+    if (!NON_PROXY_HOST_PATTERN.matcher(value).matches()) {
```

Comment:
since this is the last check, could replace these last few lines with:
`return NON_PROXY_HOST_PATTERN.matcher(value).matches();`